Isosurfacing_3/include/CGAL/Isosurfacing_3/internal/Marching_cubes_3_internal.h

```diff
@@ -0,0 +1,213 @@
+    std::size_t v_id = 0;
+    std::bitset<Domain_::VERTICES_PER_CELL> index = 0;
+    for (const Vertex_handle& v : domain.cell_vertices(cell)) {
+        // collect scalar values and computex index
+        corners[v_id] = domain.position(v);
+        values[v_id] = domain.value(v);
+
+        if (values[v_id] >= iso_value) {
+            index.set(v_id);
+        }
+        // next cell vertex
+        v_id++;
+    }
+
+    return static_cast<std::size_t>(index.to_ullong());
+}
+
+template <class CellEdges, typename FT, typename Corners_, typename Values_, typename Vertices_>
+void mc_construct_vertices(const CellEdges& cell_edges, const FT iso_value, const std::size_t i_case,
+                           const Corners_& corners, const Values_& values, Vertices_& vertices) {
+
+    // compute for this case the vertices
+    std::size_t flag = 1;
+    std::size_t e_id = 0;
+
+    for (const auto& edge : cell_edges) {
+        (void)edge;  // TODO
+
+        if (flag & Cube_table::intersected_edges[i_case]) {
+
+            // generate vertex here, do not care at this point if vertex already exist
+
+            const int v0 = Cube_table::edge_to_vertex[e_id][0];
+            const int v1 = Cube_table::edge_to_vertex[e_id][1];
+
+            vertices[e_id] = vertex_interpolation(corners[v0], corners[v1], values[v0], values[v1], iso_value);
+        }
+        flag <<= 1;
+        e_id++;
+    }
+}
+
+template <typename Vertices_, class TriangleList>
+void mc_construct_triangles(const int i_case, const Vertices_& vertices, TriangleList& triangles) {
+    // construct triangles
+    for (int t = 0; t < 16; t += 3) {
+
+        const int t_index = i_case * 16 + t;
+        // if (e_tris_list[t_index] == 0x7f)
+        if (Cube_table::triangle_cases[t_index] == -1) break;
+
+        const int eg0 = Cube_table::triangle_cases[t_index + 0];  // TODO: move more of this stuff into the table
+        const int eg1 = Cube_table::triangle_cases[t_index + 1];
+        const int eg2 = Cube_table::triangle_cases[t_index + 2];
+
+        // insert new triangle in list
+        std::array<Point, 3> points;
+        points[0] = vertices[eg0];
+        points[1] = vertices[eg1];
+        points[2] = vertices[eg2];
+
+        triangles.push_back(points);
+    }
+}
+
+template <class TriangleList, class PointRange, class PolygonRange>
+void to_indexed_face_set(const TriangleList& triangle_list, PointRange& points, PolygonRange& polygons) {
+    for (auto& triangle : triangle_list) {
+        const std::size_t id = points.size();
+
+        points.push_back(triangle[0]);
+        points.push_back(triangle[1]);
+        points.push_back(triangle[2]);
+
+        polygons.push_back({});
+        auto& triangle = polygons.back();
+        triangle.push_back(id + 2);
+        triangle.push_back(id + 1);
+        triangle.push_back(id + 0);
+    }
+}
+
+template <class Domain_>
+class Marching_cubes_functor {
+private:
+    typedef Domain_ Domain;
```

Comment:
@sloriot can you advice if instead of `typedef` new packages should systematically use `using`?